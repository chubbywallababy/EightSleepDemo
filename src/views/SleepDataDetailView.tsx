import React, {useCallback, useEffect, useState} from 'react';
import {ScrollView, StyleSheet, Text, View} from 'react-native';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RootStackParamList} from '../../App';
import {
  getHeartRates,
  getMovements,
  getRespirations,
  getSortedStages,
} from '../utils/SleepDataUtils';
import {SleepDataByStageChart} from '../components/SleepDataByStageChart';
import {getLineData} from '../components/utils/getLineData';
import {SleepStageSwitch} from '../components/SleepStageSwitch';
import {Stage} from '../types';
import {SleepDataByLineItem} from '../types/SleepDataByLineItem';
import {SleepStagesPieGraph} from '../components/SleepStagesPieGraph';

type SleepDataDetailViewProps = NativeStackScreenProps<
  RootStackParamList,
  'Details'
>;

export const SleepDataDetailView = ({
  route,
  navigation,
}: SleepDataDetailViewProps) => {
  const [selectedStage, setSelectedStage] = useState<Stage>(Stage.Deep);

  const data = route.params.data;

  useEffect(() => {
    navigation?.setOptions({title: route.params.data.date});
  }, [navigation, route.params.data.date]);

  const getCurrentField = useCallback(
    (givenData: SleepDataByLineItem) => {
      switch (selectedStage) {
        case Stage.Deep:
          return givenData.deep;
        case Stage.Light:
          return givenData.light;
        case Stage.Rem:
          return givenData.rem;
        default:
          throw new Error('Encountered new stage: ' + selectedStage);
      }
    },
    [selectedStage],
  );

  const heartData = getHeartRates(data);
  const moveData = getMovements(data);
  const respData = getRespirations(data);

  const tempData = getLineData(getSortedStages(data), 'temperature');

  return (
    <View>
      <View style={styles.switchContainer}>
        <SleepStageSwitch stage={selectedStage} setStage={setSelectedStage} />
      </View>
      <ScrollView>
        <SleepStagesPieGraph />
        <SleepDataByStageChart
          data={getCurrentField(tempData)}
          title="Temperature"
          units={'Celcius'}
          yAxisOffset={30}
          spacing={22}
          noOfSections={3}
          initialSpacing={10}
        />
        <Text style={styles.textField}>{JSON.stringify(tempData)}</Text>
        <Text style={styles.textField}>{JSON.stringify(heartData)}</Text>
        <Text style={styles.textField}>{JSON.stringify(moveData)}</Text>
        <Text style={styles.textField}>{JSON.stringify(respData)}</Text>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  textField: {
    padding: 20,
  },
  switchContainer: {
    paddingTop: 10,
  },
});
