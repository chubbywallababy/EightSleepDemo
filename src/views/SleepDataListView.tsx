// import React, {useEffect, useState} from 'react';
// import {
//   View,
//   StyleSheet,
//   FlatList,
//   Text,
//   ActivityIndicator,
// } from 'react-native';
// import {useAppDispatch, useAppSelector} from '../redux/hooks';
// import {fetchAsync, selectData, selectStatus} from '../redux/sleeper/sleepSlice';
// import {SleepDataCell} from '../components/SleepDataCell';
// import {useNavigation} from './hooks/useNavigation';
// import {AnimatedNumber} from '../components/AnimatedNumber';
// import {Background} from '../components/common';

// export const SleepDataListView = () => {
//   const navigation = useNavigation();

//   const [didLoad, setDidLoad] = useState(false);
//   const data = useAppSelector(selectData);
//   const status = useAppSelector(selectStatus);

//   const dispatch = useAppDispatch();

//   useEffect(() => {
//     if (!didLoad) {
//       dispatch(fetchAsync());
//       setDidLoad(true);
//     }
//   }, [dispatch, didLoad]);

//   return (
//     <Background style={styles.container}>
//       {status === 'loading' ? <ActivityIndicator size="large" /> : null}
//       {status === 'failed' ? <Text>Sorry, something went wrong...</Text> : null}
//       {data.length === 0 && status === 'idle' ? (
//         <Text>There are no nights available for analysis</Text>
//       ) : null}
//       {data.length > 0 && status === 'idle' ? (
//         <>
//           <View style={styles.heading}>
//             <AnimatedNumber n={data.length} duration={500} />
//             <Text style={styles.headingText}>nights tracked</Text>
//           </View>
//           <FlatList
//             style={styles.list}
//             keyExtractor={item => item.id}
//             data={data}
//             renderItem={({item}) => (
//               <SleepDataCell
//                 data={item}
//                 onPress={() => navigation?.navigate('Details', {data: item})}
//               />
//             )}
//           />
//         </>
//       ) : null}
//     </Background>
//   );
// };

// const styles = StyleSheet.create({
//   container: {
//     flex: 1,
//     width: '100%',
//     justifyContent: 'center',
//     alignItems: 'center',
//   },
//   paddedText: {
//     marginVertical: 100,
//   },
//   heading: {
//     paddingVertical: 20,
//     display: 'flex',
//     flexDirection: 'row',
//     alignItems: 'center',
//   },
//   headingText: {
//     marginLeft: 10,
//     fontSize: 16,
//   },
//   list: {
//     flex: 1,
//     width: '100%',
//   },
// });
