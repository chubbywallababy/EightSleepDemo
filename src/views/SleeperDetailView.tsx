import React, {useCallback, useEffect, useState} from 'react';
import {ScrollView, StyleSheet, Text, View} from 'react-native';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RootStackParamList} from '../../App';
import {
  getHeartRates,
  getMovements,
  getRespirations,
  getSortedStages,
} from '../utils/SleepDataUtils';
import {SleepDataByStageChart} from '../components/SleepDataByStageChart';
import {getLineData} from '../components/utils/getLineData';
import {SleepStageSwitch} from '../components/SleepStageSwitch';
import {Stage} from '../types';
import {SleepDataByLineItem} from '../types/SleepDataByLineItem';
import {SleepStagesPieGraph} from '../components/SleepStagesPieGraph';
import {Background} from '../components/common';

type SleepDataDetailViewProps = NativeStackScreenProps<
  RootStackParamList,
  'Details'
>;

export const SleeperDetailView = ({
  route,
  navigation,
}: SleepDataDetailViewProps) => {
  const [selectedStage, setSelectedStage] = useState<Stage>(Stage.Deep);

  useEffect(() => {
    navigation?.setOptions({title: route.params.data.name});
  }, [navigation, route.params.data.name]);

  return (
    <Background>
      <View style={styles.switchContainer}>
        <SleepStageSwitch stage={selectedStage} setStage={setSelectedStage} />
      </View>
    </Background>
  );
};

const styles = StyleSheet.create({
  textField: {
    padding: 20,
  },
  switchContainer: {
    paddingTop: 10,
  },
});
