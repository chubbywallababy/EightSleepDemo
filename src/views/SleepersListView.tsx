import React, {useEffect, useState} from 'react';
import {
  StyleSheet,
  FlatList,
  Text,
  ActivityIndicator,
  View,
} from 'react-native';
import {useAppDispatch, useAppSelector} from '../redux/hooks';
import {useNavigation} from './hooks/useNavigation';
import {Background, SleepText, SleepView} from '../components/common';
import {
  selectUsers,
  selectUsersError,
  selectUsersStatus,
} from '../redux/users/selectors';
import {SleeperCell} from '../components/SleeperCell';
import {usersActions} from '../redux/users/actions';

export const SleepersListView = () => {
  const navigation = useNavigation();

  const [didLoad, setDidLoad] = useState(false);
  const data = useAppSelector(selectUsers);
  const status = useAppSelector(selectUsersStatus);
  const error = useAppSelector(selectUsersError);

  const dispatch = useAppDispatch();

  useEffect(() => {
    if (!didLoad) {
      dispatch(usersActions.fetchUsers());
      setDidLoad(true);
    }
  }, [dispatch, didLoad]);

  return (
    <Background style={styles.container}>
      {status === 'loading' ? <ActivityIndicator size="large" /> : null}
      {status === 'failed' ? (
        <SleepView>
          <SleepText>Sorry, something went wrong...</SleepText>
          {error ? <SleepText>{error}</SleepText> : null}
        </SleepView>
      ) : null}
      {data.length === 0 && status === 'idle' ? (
        <Text>There are no users available for analysis</Text>
      ) : null}
      {data.length > 0 && status === 'idle' ? (
        <FlatList
          style={styles.list}
          keyExtractor={item => item.id}
          data={data}
          renderItem={({item}) => (
            <SleeperCell
              data={item}
              onPress={() => navigation?.navigate('Details', {data: item})}
            />
          )}
          ItemSeparatorComponent={Separator}
        />
      ) : null}
    </Background>
  );
};

const Separator = () => <View style={styles.separator} />;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  list: {
    flex: 1,
    width: '100%',
    marginVertical: 20,
  },
  separator: {height: 12},
});
