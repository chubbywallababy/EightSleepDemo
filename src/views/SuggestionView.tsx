import React, {useCallback} from 'react';
import {TouchableOpacity, View} from 'react-native';
import {CardTitle, SleepText} from '../components/common';
import {strings} from '../i18n';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RootStackParamList} from '../../App';
import {StyleSheet} from 'react-native';
import {colors} from '../styles/colors';
import {useAppDispatch} from '../redux/hooks';
import {acceptSuggestion, denySuggestion} from '../redux/users/slice';

type SuggestionViewProps = NativeStackScreenProps<
  RootStackParamList,
  'Suggestion'
>;

/**
 * A view for a user to view and accept suggestions. This is currently only a view with text and buttons.
 *
 * Future improvements would be to include a graph like the one in the mocks
 */
export const SuggestionView = ({route, navigation}: SuggestionViewProps) => {
  const dispatch = useAppDispatch();

  const makeSelection = useCallback(
    (accept: boolean) => {
      if (accept) {
        dispatch(acceptSuggestion(route.params.data.id));
      } else {
        dispatch(denySuggestion(route.params.data.id));
      }
      navigation.goBack();
    },
    [dispatch, route, navigation],
  );

  return (
    <View style={styles.container}>
      <View style={styles.cardContainer}>
        <CardTitle style={[styles.text, styles.title]}>
          {strings.suggestion.title}
        </CardTitle>
        <SleepText style={[styles.text, styles.suggestionText]}>
          {strings.suggestion.text}
        </SleepText>
      </View>
      <View style={styles.buttonContainer}>
        <TouchableOpacity
          onPress={() => makeSelection(true)}
          style={[styles.button, styles.primaryButton]}>
          <SleepText style={styles.buttonText}>
            {strings.common.acceptChanges}
          </SleepText>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => makeSelection(false)}
          style={styles.button}>
          <SleepText style={styles.buttonText}>
            {strings.common.notTonight}
          </SleepText>
        </TouchableOpacity>
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    padding: 8,
    gap: 20,
  },
  cardContainer: {
    padding: 8,
    gap: 12,
  },
  text: {
    textAlign: 'center',
  },
  suggestionText: {
    fontSize: 16,
  },
  title: {
    textTransform: 'capitalize',
  },
  buttonText: {
    fontSize: 14,
    fontWeight: '700',
  },
  button: {
    height: 54,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 6,
  },
  primaryButton: {
    backgroundColor: colors.primary,
  },
  buttonContainer: {
    gap: 8,
  },
});
