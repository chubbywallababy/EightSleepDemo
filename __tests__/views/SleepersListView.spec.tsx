import React from 'react';
import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import {SleepersListView} from '../../src/views/SleepersListView';
import {mockStore} from '../__mocks__/mockStore';
import {TouchableOpacity} from 'react-native';

jest.mock('../../src/components/SleepLinearGradient', () => ({
  SleepLinearGradient: ({children}: {children: React.ReactNode}) => (
    <>{children}</>
  ),
}));

jest.mock('../../src/components/GlowingBorder', () => ({
  GlowingBorder: ({children}: {children: React.ReactNode}) => <>{children}</>,
}));

jest.mock('../../src/components/AnimatedNumber', () => ({
  AnimatedNumber: () => <></>,
  numberStyles: {text: {}},
}));

describe('boilerplate for SleepersListView', () => {
  it('should render the view', async () => {
    const {root, toJSON} = render(
      <Provider store={mockStore}>
        <SleepersListView />
      </Provider>,
    );

    const view = root.findAllByType(TouchableOpacity);
    expect(view).toHaveLength(3);

    expect(toJSON()).toMatchSnapshot();
  });
});
