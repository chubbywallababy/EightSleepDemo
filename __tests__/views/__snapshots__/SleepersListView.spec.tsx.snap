// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boilerplate for SleepersListView should render the view 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#000",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      [
        {
          "email": "doejohn@email.com",
          "id": "2228b530e055401f81ba37b51ff6f81d",
          "name": "John Doe",
        },
        {
          "email": "doejane@email.com",
          "id": "d6c1355e38194139b8d0c870baf86365",
          "name": "Jane Doe",
        },
        {
          "email": "doejim@email.com",
          "id": "f9bf229fd19e4c799e8c19a962d73449",
          "name": "Jim Doe",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      {
        "flex": 1,
        "marginVertical": 20,
        "width": "100%",
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
              "marginHorizontal": 0,
              "opacity": 1,
            }
          }
          testID="2228b530e055401f81ba37b51ff6f81d"
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#202020",
                },
                {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "flexDirection": "column",
                  "height": 125,
                  "justifyContent": "space-between",
                  "padding": 12,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "white",
                    },
                    {
                      "fontSize": 20,
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                John Doe
              </Text>
              <View
                style={
                  {
                    "paddingHorizontal": 18,
                    "paddingVertical": 5,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "white",
                      },
                      {
                        "fontSize": 12,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  See insights
                </Text>
              </View>
              <Image
                source={
                  {
                    "testUri": "../../../src/components/images/Chevron.png",
                  }
                }
                style={
                  [
                    {
                      "height": 16,
                      "width": 9,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                  {
                    "paddingHorizontal": 15,
                  },
                ]
              }
            >
              <View>
                <Text
                  style={
                    {
                      "color": "#ADB3C5",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "paddingBottom": 5,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Deep sleep
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "white",
                      },
                      {},
                    ]
                  }
                >
                  1h 2m
                </Text>
              </View>
              <View>
                <Text
                  style={
                    {
                      "color": "#ADB3C5",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "paddingBottom": 5,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Sleep score
                  <Text
                    style={
                      {
                        "fontSize": 11,
                        "textTransform": "lowercase",
                      }
                    }
                  >
                     (%)
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "height": 12,
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
              "marginHorizontal": 0,
              "opacity": 1,
            }
          }
          testID="d6c1355e38194139b8d0c870baf86365"
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#202020",
                },
                {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "flexDirection": "column",
                  "height": 125,
                  "justifyContent": "space-between",
                  "padding": 12,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "white",
                    },
                    {
                      "fontSize": 20,
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                Jane Doe
              </Text>
              <Image
                source={
                  {
                    "testUri": "../../../src/components/images/Chevron.png",
                  }
                }
                style={
                  [
                    {
                      "height": 16,
                      "width": 9,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                  {
                    "paddingHorizontal": 15,
                  },
                ]
              }
            >
              <View>
                <Text
                  style={
                    {
                      "color": "#ADB3C5",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "paddingBottom": 5,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Deep sleep
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "white",
                      },
                      {},
                    ]
                  }
                >
                  1h 52m
                </Text>
              </View>
              <View>
                <Text
                  style={
                    {
                      "color": "#ADB3C5",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "paddingBottom": 5,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Sleep score
                  <Text
                    style={
                      {
                        "fontSize": 11,
                        "textTransform": "lowercase",
                      }
                    }
                  >
                     (%)
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "height": 12,
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
              "marginHorizontal": 0,
              "opacity": 1,
            }
          }
          testID="f9bf229fd19e4c799e8c19a962d73449"
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#202020",
                },
                {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "flexDirection": "column",
                  "height": 125,
                  "justifyContent": "space-between",
                  "padding": 12,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "white",
                    },
                    {
                      "fontSize": 20,
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                Jim Doe
              </Text>
              <Image
                source={
                  {
                    "testUri": "../../../src/components/images/Chevron.png",
                  }
                }
                style={
                  [
                    {
                      "height": 16,
                      "width": 9,
                    },
                    undefined,
                  ]
                }
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                  {
                    "paddingHorizontal": 15,
                  },
                ]
              }
            >
              <View>
                <Text
                  style={
                    {
                      "color": "#ADB3C5",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "paddingBottom": 5,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Deep sleep
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "white",
                      },
                      {},
                    ]
                  }
                >
                  2h 14m
                </Text>
              </View>
              <View>
                <Text
                  style={
                    {
                      "color": "#ADB3C5",
                      "fontSize": 12,
                      "fontWeight": "400",
                      "paddingBottom": 5,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Sleep score
                  <Text
                    style={
                      {
                        "fontSize": 11,
                        "textTransform": "lowercase",
                      }
                    }
                  >
                     (%)
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
